### PR TITLE
RavenDB-14482 Prevent from putting reference documents if generated pattern ID is invalid (if it ends with / or | or contains null field values)

### PR DESCRIPTION
**Pull request by arekpalinski**
https://issues.hibernatingrhinos.com/issue/RavenDB-14482

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReferencesPattern.cs [hunk 1]
```diff
@@ -105,7 +107,20 @@ public string GetId()
...
+                var id = _id.AppendFormat(_formattedPattern, _values).ToString();
+                
+                ValidateId(id);
+
+                return id;
+            }
+
+            public static void ValidateId(string id)
+            {
+                if (id.EndsWith('/'))
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReferencesPattern.cs:119]: After merging this to 5.0 we need to use DocumentDatabase.IndentityPartsSeparator here

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReduceToCollectionCommand.cs [hunk 2]
```diff
@@ -145,16 +145,30 @@ private BlittableJsonReaderObject GenerateReduceOutput(object reduceObject, Inde
...
+                        try
+                        {
+                            referenceDocIdBuilder.Add(property.Key, property.Value);
+                        }
+                        catch (Exception e)
+                        {
+                            patternPropertiesAddedSuccessfully = false;
+
+                            if (stats != null) // should never be null
+                                stats.AddReduceError($"Failed to build document ID based on provided pattern for output to collection references. {e.Message}");
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReduceToCollectionCommand.cs]: Can we add the full exception details?

Comment (arekpalinski) [reply src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReduceToCollectionCommand.cs]: Fixed in https://github.com/ravendb/ravendb/pull/10285/commits/14b715f1b78381ca8b839ca027e8fd7c57e4ee2e

File: src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReferencesPattern.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReferencesPattern.cs:119]: Okay. Added TODO in issue comment

**Pull request merged**